Make "Display hidden components in Viewer" checkbox screen-dependent

matthewcoufal opened the pull request:
This PR allows for the visibility of screen components to be toggled on a screen basis. That is, it adjusts the checkbox so that its state change only affects the current screen, not all screens in the project.

AppInventorWorkerBee commented:
Can one of the admins verify this patch?

ewpatton commented:
@AppInventorWorkerBee ok to test

matthewcoufal commented:
@ewpatton I moved the maintenance of the checkbox state to `YaFormEditor`, and you were right - it is much simpler than what I had been doing. Thanks for the tips!

ewpatton commented on appinventor/appengine/src/com/google/appinventor/client/editor/ProjectEditor.java:
```diff
@@ -7,6 +7,7 @@
 package com.google.appinventor.client.editor;
 
 import com.google.appinventor.client.Ode;
+import com.google.appinventor.client.editor.simple.components.MockComponent;
```
@matthewcoufal You can remove this import now that the ret of the changes have been removed from this file.

ewpatton commented on appinventor/appengine/src/com/google/appinventor/client/editor/simple/SimpleVisibleComponentsPanel.java:
```diff
@@ -35,7 +36,7 @@
   // UI elements
   private final VerticalPanel phoneScreen;
-  private final CheckBox checkboxShowHiddenComponents;
+  public final CheckBox checkboxShowHiddenComponents;
```
Rather than making this public, add a getter to retrieve the information you need.

ewpatton commented on appinventor/appengine/src/com/google/appinventor/client/editor/simple/components/MockComponent.java:
```diff
@@ -523,6 +524,10 @@ public final void addProperty(String name, String defaultValue, String caption,
     properties.addProperty(name, defaultValue, caption, editor, type);
   }
 
+  public SimpleEditor getEditor() {
```
I still think my original claim here is true.

ewpatton commented on appinventor/appengine/src/com/google/appinventor/client/editor/youngandroid/YaFormEditor.java:
```diff
@@ -187,6 +187,10 @@ String getFileContent() {
     setSize("100%", "100%");
   }
 
+  public boolean shouldDisplayHiddenComponents() {
```
See my other note about a getter for the state of the checkbox without exposing the underlying object model.

ewpatton commented on appinventor/appengine/src/com/google/appinventor/client/editor/simple/SimpleVisibleComponentsPanel.java:
```diff
@@ -64,23 +65,11 @@ public SimpleVisibleComponentsPanel(final SimpleEditor editor,
     };
     checkboxShowHiddenComponents.addValueChangeHandler(new ValueChangeHandler<Boolean>() {
       @Override
       public void onValueChange(ValueChangeEvent<Boolean> event) {
```
One problem with removing this functionality is that we will no longer save state between sessions.

matthewcoufal commented:
@SusanRatiLane Can you take a look at my most recent changes? After testing, I think I've enabled the ability to save state between sessions with these changes, and I have addressed above comments.
